### PR TITLE
passt..detach: wait for guest os booting before detaching

### PR DESCRIPTION
**Pull request by yafu-1**

### CODE REVIEW

**Comment by yafu-1** [comment]
Test results:
$/usr/local/bin/avocado run --vt-type libvirt --test-runner=runner --vt-machine-type arm64-mmio virtual_network.passt.attach_detach.ip_portfw.detach.root_user --vt-connect-uri qemu:///system
No python imaging library installed. Screendump and Windows guest BSOD detection are disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
No python imaging library installed. PPM image conversion to JPEG disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
JOB ID     : a9ed80d9a3201626cb659724f629abc9568ac8c4
JOB LOG    : /var/lib/avocado/job-results/job-2024-03-12T23.18-a9ed80d/job.log
 (1/1) type_specific.io-github-autotest-libvirt.virtual_network.passt.attach_detach.ip_portfw.detach.root_user: PASS (65.75 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /var/lib/avocado/job-results/job-2024-03-12T23.18-a9ed80d/results.html

**Comment by yafu-1** [comment]
@chloerh @dzhengfy Could you help to review the pr please? Thanks.

**Comment by dzhengfy** [comment]
@yafu-1 For fixing an issue, please add the test result before your fix which will help reviewer know what the problem you intend to solve. Thanks

File: libvirt/tests/src/virtual_network/passt/passt_attach_detach.py [hunk 1]
```diff
@@ -123,7 +123,7 @@ def run(test, params, env):
             if not os.path.exists(log_file):
                 test.fail(f'Logfile of passt "{log_file}" not created')
 
-            session = vm.wait_for_serial_login(timeout=60)
+            session = vm.wait_for_serial_login(timeout=60).close()
```
Comment (dzhengfy) [inline libvirt/tests/src/virtual_network/passt/passt_attach_detach.py]: What is the reason to close the session? The session will be used in next line.

**Review by chloerh — APPROVED** [review]

File: libvirt/tests/src/virtual_network/passt/passt_attach_detach.py [hunk 2]
```diff
@@ -155,6 +155,8 @@ def run(test, params, env):
             LOG.debug(f'vmxml before detach:\n{vmxml}')
             iface_to_detach = vmxml.get_devices('interface')[0]
             mac = iface_to_detach.mac_address
+            #wait for guest os to boot completely before detaching interface
```
[the remark this answers is not in the document]
Comment (yafu-1) [reply libvirt/tests/src/virtual_network/passt/passt_attach_detach.py]: Updated.

File: libvirt/tests/src/virtual_network/passt/passt_attach_detach.py (hunk 1 above)
Comment (yafu-1) [reply libvirt/tests/src/virtual_network/passt/passt_attach_detach.py]: It's a mistake, I added close() to wrong session. Have updated.

**Review by chunfuwen — APPROVED** [review]
lgtm

**Review by dzhengfy — APPROVED** [review]

**Pull request merged**